crates/tabby-common/src/api/code.rs

```diff
@@ -38,6 +38,12 @@ pub enum CodeSearchError {
     TantivyError(#[from] tantivy::TantivyError),
 }
 
+#[derive(Error, Debug)]
+pub enum CodeCompletionError {
```

Comment:
this should be in serve/completions.rs

Reply:
Noted.

But I also noticed there was a refactoring PR which removed `thiserror` dependency ([here](https://github.com/TabbyML/tabby/pull/743/files#diff-776f8f8254316885ba3ec0e58a6e59829cce073ab65ee189061c32506566ad4cL45)) from `tabby` crate.

I probably need to add it back since this `enum` depends on `Error` macro from `thiserror` lib if I put it in `tabby` crate, is it ok?

Or, this enum still resides in `tabby-common` crate, but in separate module file?

Reply:
adding thiserror back to`tabby`is good